File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.java
import java.util.*;

class Solution {
    public int[] solution(String[] keymap, String[] targets) {
        // 구현
        // target 한바퀴 도는 동안 keymap에서 찾아야함
        // keymap hashmap으로 하면 효율 괜찮을듯
        // keymap setting 시 최소값으로 저장
        
        HashMap<Character, Integer> map = new HashMap<>();
        
        int tarLen = targets.length;
        int mapLen = keymap.length;
        int[] answer = new int[tarLen];
        
        // setting
        for (int i = 0; i < mapLen; i++) {
            for (int j = 0; j < keymap[i].length(); j++) {
                char curr = keymap[i].charAt(j);
                
                if (map.containsKey(curr)) {
                    int idx = map.get(curr);
                    map.put(curr, Math.min(j + 1, idx));
                } else {
                    map.put(curr, j + 1);
                }
            }
        }
        
        // logic
        for (int i = 0; i < tarLen; i++) {
            String target = targets[i];
            int count = 0;
            boolean flag = true;
            
            for (char curr : target.toCharArray()) {
                if (map.containsKey(curr)) {
                    count += map.get(curr);
                } else {
                    flag = false;
                    break;
                }
            }
            
            answer[i] = flag == false ? -1 : count;
        }
        
        
        
        return answer;
    }
}

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.cs
using System;
using System.Collections.Generic;

public class Solution {
    public int[] solution(string[] keymap, string[] targets) {
        
        Dictionary<char, int> map = new Dictionary<char, int>();
        int tarLen = targets.Length;
        int mapLen = keymap.Length;
        int[] answer = new int[tarLen];
        
        for (int i = 0; i < mapLen; i++) 
        {
            for (int j = 0; j < keymap[i].Length; j++)
            {
                char curr = keymap[i][j];
                
                if (map.ContainsKey(curr))
                {
                    int idx = map[curr];
                    map[curr] = Math.Min(j + 1, idx);
                }
                else 
                {
                    map[curr] = j + 1;
                }
            }
        }
        
        for (int i = 0; i < tarLen; i++) 
        {
            string target = targets[i];
            int count = 0;
            
            foreach(char cur in target)
            {
                if (map.ContainsKey(cur))
                {
                    count += map[cur];
                }
                else 
                {
                    count = -1;
                    break;
                }
            }
            
            answer[i] = count;
        }
        
        return answer;
    }
}